### ISSUE TITLE
Unable to use System.Drawing.common library with gauge

### ISSUE THREAD

**Issue by pp1609** [getgauge/gauge-dotnet#49]
Hi,

I have a requirement to use Methods in System.Drawing.common library methods. In simple dotnet core project it is working but with Gauge the same is not working and producing following error:
![Error](https://user-images.githubusercontent.com/36397242/67073304-70bee500-f1a4-11e9-8267-fbe6928a839b.png)

Thanks,
Preeti

**Comment by nehashri** [comment]
@pp1609 Is there any reason to use `System.Drawing.common` library instead of others like [ImageSharp](https://github.com/SixLabors/ImageSharp?WT.mc_id=-blog-scottha)? 
The issue is most likely due to gauge loading the step implementations for execution.

**Comment by pp1609** [comment]
Yes i can use ImageSharp for image processing but i need some way to capture the desktop screenshot in my automation, and didn't find any way for that other then System.Drawing.common library.

**Comment by shubhamsc** [comment]
@pp1609 Did you check the taking screenshot with Gauge?
you can capture the desktop screenshot with gauge by adding this line `GaugeScreenshots.Capture();` in your implementation code. and you can customize the behaviour of taking a screenshot with the [custom screenshot](https://docs.gauge.org/writing-specifications.html?os=windows&language=csharp&ide=visualstudio#taking-custom-screenshots).

**Comment by sriv** [comment]
So, I've been looking at this for the last two days. My suspicion originally was that perhaps gauge-dotnet was not probing the satellite assemblies properly. I don't think that is the case here.

The problem seems specific to `System.Drawing` and it's availability on various platforms. Since the `System.Drawing` namespace exposes GDI+ functionality, it also requires to make some system calls, which is not available in all platforms. [This StackOverflow answer](https://stackoverflow.com/a/51907024) has some details.

In short, it is recommended to use alternatives to `System.Drawing`. 

On that note, if you notice other dependencies/references working well, it probably eliminates the theory that gauge-dotnet is not loading assemblies properly.

**Comment by sriv** [comment]
I'll mark this issue as closed for now, as I cannot see anything that can be done as part of gauge-dotnet to address this. Please comment to let us know, if otherwise.
